storage/StorageManager.cpp

```diff
@@ -461,13 +473,18 @@ MutableBlockReference StorageManager::getBlockInternal(
   // doesn't know about the block until blockReferenced is called, so
   // chooseBlockToEvict shouldn't return the block.
   if (!ret.valid()) {
+    // Call a best-effort method to evict blocks until the size of our buffer
+    // pool falls below the current buffer pool size plus the size of the
+    // block we are going to retrieve.
+    makeRoomForBlock(num_slots);
```

Comment:
This line should move after `Line 409`, as this is another check for whether the block has been loaded.